main.js

```diff
@@ -190,6 +207,26 @@ define(function (require, exports, module) {
 	addIcon('d', 'ion-contrast', '#960000');
 	addIcon('r', 'ion-ios-analytics', '#8495C0');
 
+	//Look up preferences and apply changes
+	function applyUserPrefs() {
```

Comment:
Don't fill the `icons` object with user icons. With this approach, reloading settings will fail when the user removed an extension from the settings file. You should check in the provider function (line 247) whether the icon exists in the `userIcons` object, and otherwise check it in the `icons` object.